File: TASC-Intern-Batch9/pharmacy/src/main/java/com/project/pharmacy/service/OrderService.java
package com.project.pharmacy.service;

import com.project.pharmacy.dto.request.OrderFilterRequest;
import com.project.pharmacy.dto.request.OrderRequest;
import com.project.pharmacy.dto.response.ApiResponse;
import com.project.pharmacy.dto.response.OrderDetailResponse;
import com.project.pharmacy.dto.response.OrderResponse;
import com.project.pharmacy.dto.response.PageResponse;

import java.util.List;

public interface OrderService {
    ApiResponse<PageResponse<List<OrderResponse>>> getAllOrders(int pageIndex, int pageSize, OrderFilterRequest request);
    ApiResponse<PageResponse<List<OrderResponse>>> getMyOrders(int pageIndex, int pageSize, String status);
    ApiResponse<List<OrderDetailResponse>> getOrderDetail(Long orderId);
    ApiResponse<?> createOrder(OrderRequest request);
    ApiResponse<OrderResponse> changeOrderStatus(Long id, String status);
    ApiResponse<OrderResponse> changePaymentStatus(Long id, String paymentStatus);
    ApiResponse<Long> getTotalOrder();

    ApiResponse<Long> getAllRevenue();

    ApiResponse<List<OrderResponse>> getFiveNewestOrder();
    ApiResponse<Void> cancelOrder(Long id);
}

File: pharmacy-v1/fe/fe-client/src/api/orderService.ts
import axiosClient from './axiosClient';
import type { ApiResponse, Order, PageResponse} from '../types';
import type { OrderRequest, OrderResponse, OrderDetailResponse } from '../types/order.types';


const orderService = {
  createOrder: (data: OrderRequest) => {
    return axiosClient.post<ApiResponse<{ paymentUrl?: string; orderId: number }>>('/orders', data);
  },

  getMyOrders: (pageIndex: number = 1, status?: string) => {
    const params: any = { pageIndex, pageSize: 10 };
    console.log("status")
    if (status && status !== 'ALL') {
      params.status = status;
    }
    return axiosClient.get<ApiResponse<PageResponse<Order>>>('/orders/my-orders', { params });
  },

  getOrderDetail: (id: number) => {
    return axiosClient.get<ApiResponse<OrderDetailResponse[]>>(`/orders/detail/${id}`);
  },

  cancelOrder: (id: number) => {
    return axiosClient.put<ApiResponse<void>>(`/orders/cancel/${id}`);
  },

  
};

export default orderService;